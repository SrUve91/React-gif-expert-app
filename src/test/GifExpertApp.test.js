import { shallow } from 'enzyme';
import React from 'react';
import { GiftExpertApp } from '../GiftExpertApp';

describe('Pruebas en GifExpertApp', () => {
   test('Se debe mostrar correctamente', () => {
      const wrapper = shallow(<GiftExpertApp/>);
      
      expect(wrapper).toMatchSnapshot();
   });

   test('Debe de mostrar una lista de categorias', () => {
      const categories = ['One punch', 'Dragon Ball'];
      const wrapper = shallow(<GiftExpertApp defaultCategories={categories}/>);

      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find('GifGrid').length).toBe(categories.length);
   })
   
});
