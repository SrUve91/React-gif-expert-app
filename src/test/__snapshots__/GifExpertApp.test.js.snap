// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp Debe de mostrar una lista de categorias 1`] = `
<div>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One punch"
      key="One punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</div>
`;

exports[`Pruebas en GifExpertApp Se debe mostrar correctamente 1`] = `
<div>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;
